servers/rendering/renderer_rd/storage_rd/light_storage.h

```diff
@@ -544,6 +544,29 @@ class LightStorage : public RendererLightStorage {
 		return light->shadow;
 	}
 
+	virtual bool light_is_shadow_visible_at_position(RID p_light_instance, const Vector3 position) const override {
```

Comment:
Most of the functions here take a light RID directly, you may be able to do this by passing `ins->base` for the light instance from `renderer_scene_cull`.

Reply:
How do I get the position from the light?

Reply:
Ah I see what you mean, you would have to pass the light position, so I guess it's not too bad as is.

Although I've realised if you are passing light instance, it should be in the `LIGHT INSTANCE API` section below, and the func name should start with `light_instance` rather than `light`.

I also wonder whether `light_instance_is_shadow_visible_at_camera_position()` might be slightly less confusing than just `position`, as it is hard to interpret what the function does? :thinking: Naming is always really difficult.

_(Another alternative would be to query the threshold distance from the light `RID` and do the calculation in the calling code.)_

Reply:
Oh, maybe.  Originally, I was trying to do it with just the light, but it seemed I needed the light instance to get the position.

I was also going to do the calculation in the calling code but realized that most of the time the distance is never needed since lights don't have the distance fade enabled by default, so it's probably more optimal to only do the distance calculation when actually needed.